### PR TITLE
Core: Fix meta.yaml and allow the `None` game category for common options

### PR DESCRIPTION
**Pull request by alwaysintreble**

### CODE REVIEW

File: Generate.py [hunk 1]
```diff
@@ -128,12 +128,12 @@ def main(args=None, callback=ERmain):
...
     if args.meta_file_path and os.path.exists(args.meta_file_path):
         try:
-            weights_cache[args.meta_file_path] = read_weights_yamls(args.meta_file_path)
+            meta_weights = read_weights_yamls(args.meta_file_path)[-1]
         except Exception as e:
             raise ValueError(f"File {args.meta_file_path} is destroyed. Please fix your yaml.") from e
-        meta_weights = weights_cache[args.meta_file_path][-1]
         print(f"Meta: {args.meta_file_path} >> {get_choice('meta_description', meta_weights)}")
-        del(meta_weights["meta_description"])
+        if "meta_description" in meta_weights:
```
Comment (Berserker66) [inline Generate.py:135]: The old requirement for a meta_description had the side-effect of ensuring it is actually a meta file. By making it optional there is now no obvious error associated with that mistake.

Comment (alwaysintreble) [reply Generate.py:135]: Is there a better way of verifying it or do you want me to revert this line?

File: Generate.py [hunk 2]
```diff
@@ -382,6 +386,28 @@ def update_weights(weights: dict, new_weights: dict, type: str, name: str) -> di
     return weights
 
 
+def roll_meta_option(option_key, game: str, category_dict: Dict) -> Any:
+    if not game:
+        return get_choice(option_key, category_dict)
+    if game in AutoWorldRegister.world_types:
+        for key, option in AutoWorldRegister.world_types[game].options.items() | Options.per_game_common_options.items():
+            if option_key == key:
```
Comment (Berserker66) [inline Generate.py:394]: slow X over Y iteration when both ends support a direct lookup should be avoived

File: Generate.py [hunk 3]
```diff
@@ -181,32 +181,36 @@ def main(args=None, callback=ERmain):
...
+    player_path_cache = {}
+    for player in range(1, args.multi + 1):
+        player_path_cache[player] = player_files.get(player, args.weights_file_path)
     name_counter = Counter()
     erargs.player_settings = {}
 
     player = 1
     while player <= args.multi:
         path = player_path_cache[player]
+        if path == args.meta_file_path:
```
Comment (Berserker66) [inline Generate.py:212]: would it not make more sense to not have have the file in the general path cache in the first place?

File: Generate.py (hunk 1 above)
Comment (Berserker66) [reply Generate.py:135]: meta_description was kind of invented for that purpose. So unless you have a better solution it's probably best to bring it back and document the why this time.

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**